File: C++/Factorial.cpp
#include<iostream>
using namespace std;

int main() {

    int num;
    cout << "Enter a non-negative number: ";
    cin >> num;

    // Ensure the number is non-negative
    if (num < 0) {
        cout << "Factorial is not defined for negative numbers." << endl;
    } else {
        int fact = 1;

        // Factorial of 0 is 1
        for (int i = num; i >= 1; i--) {
            fact = fact * i;
        }

        cout << "The factorial of " << num << " is " << fact << endl;
    }

    return 0;
}

File: DSA/Factorial.py
# Factorial.py

def factorial(n):
    """Calculate the factorial of a number recursively."""
    if n < 0:
        raise ValueError("Factorial is not defined for negative numbers.")
    elif n == 0 or n == 1:
        return 1
    else:
        return n * factorial(n - 1)

if __name__ == "__main__":
    # Example usage
    try:
        number = int(input("Enter a non-negative integer: "))
        result = factorial(number)
        print(f"The factorial of {number} is {result}.")
    except ValueError as e:
        print(e)
